customers/models.py

```diff
@@ -35,6 +38,186 @@ def get_queryset(self):
             )
         )
 
+    @transaction.atomic
+    def import_customer_data(self, data):  # noqa: C901
+        """
+        Imports list of customers of the following shape:
+        {
+            "id": "98edea83-4c92-4dda-bb90-f22e9dafe94c"  <-- profile should have this very ID
+            "comment": ""
+            "customer_id": "313431",  <-- not really used for any field
+            "boats": [
+                {
+                    "boat_type": "Perämoottorivene",  <-- Finnish name for the boat type, from translated_fields
+                    "name": "McBoatface 111",
+                    "registration_number": "31123A",
+                    "width": "2.00",
+                    "length": "5.30",
+                    "draught": null,  <-- either null or string like width and length above
+                    "weight": null  <-- either null or integer
+                }
+            ],
+            "leases": [
+                {
+                    "harbor_servicemap_id": "41074",
+                    "pier_id": "A",  <-- harbor identifier
+                    "berth_number": 4,
+                    "start_date": "2019-06-10",
+                    "end_date": "2019-09-14",
+                    "boat_index": 0  <-- index of the boat in the "boats" list
+                }
+            ],
+            "orders": [
+                {
+                    "created_at": "2019-12-02 00:00:00.000",
+                    "order_sum": "251.00",
+                    "vat_percentage": "25.0",
+                    "berth": {
+                        "harbor_id": "41074",  <-- service map id
+                        "pier_id": "A",  <-- harbor identifier
+                        "berth_number": 4
+                    },
+                    "comment": "Laskunumero: 247509 RAMSAYRANTA A 004"
+                }
+            ],
+            "organization": {  <-- either a dict like here or it will not be in this customer's dict at all
+                "type": "company",  <-- values from OrganizationType enum
+                "name": "Nice Profitable Firm Ltd.",
+                "address": "Mannerheimintie 1 A 1",
+                "postal_code": "00100",
+                "city": "Helsinki"
+            },
+        }
+
+        And returns dict where key is the customer_id and value is the UUID of created profile object
+        """
+        from leases.models import BerthLease
+        from payments.models import Order
+        from resources.models import Berth, Harbor
+
+        result = {}
+        for index, customer_data in enumerate(data):
+            try:
+                if not customer_data.get("id"):
+                    raise Exception(_("No customer ID provided"))
+
+                customer = self.create(
+                    id=customer_data.get("id"), comment=customer_data.get("comment"),
+                )
+                customer.refresh_from_db()
+
+                if organization := customer_data.get("organization"):  # noqa
```

Comment:
Is this `# noqa` comment needed?

Reply:
Yup, flake8 has trouble recognising the walrus operator, so it throws like 3 warnings.

Reply:
removed this and executed:
`flake8`
no errors 🤔

Reply:
Huh sneaky. After a bit of digging I realised that the `flake8` causing the problem was the one for `pre-commit`, which was a couple of versions behind the one we have on `requirements-dev.txt`, so by updating the flake8 version on `.pre-commit-config.yaml`, the problem is solved.